### DISCUSSION TITLE
Switch payment providers

### DISCUSSION THREAD

**Discussion by wuw-marc** [pimcore/pimcore#10453] [Q&A / Support]
I've seen a similar question in an old forum but there was no reply to it. So I'll try here.

When building a web shop I want to offer several payment options to the customer. For example I want to offer Paypal but I also want to offer credit card. Now I could do both through Paypal but for reasons I can't or don't want to. Let's just assume for this question's sake, that I do not have the option to offer two ways of payment through one provider.

When writing the config for the ecommerce framework I can configure one payment provider per tenant but not multiple. So here are my questions:

1) Is it possible to have multiple providers per tenant? And if so, how? If not, see next question.
2) Is there a way to force a different payment provider for a tenant? If not, see next question.
3) Do I understand it correctly that I'll have to set another tenant mid-checkout? How will that affect the checkout data I gathered so far?

Having the customer select their preferred way of payment before any other step in checkout is also not an option (for reasons)

Thanks in advance for any replies.

If there is no (easy) solution for what I want to achieve, I would also be grateful if you told me so ;-)

**Comment by AndrewKMoses** [comment]
Hi @wuw-marc 

Did you get any clarity on this? I'm having the same questions.

**Reply by wuw-marc** [reply]
Yes. The answer is rather simple: Change the checkout tenant mid-checkout. Since we have the same steps for every tenant and the only difference is the payment provider we haven't observed any loss of previously collected checkout data yet.

HTH, Marc

**Reply by AndrewKMoses** [reply]
Great, thanks for the reply!

**Reply by ximAABacon** [reply]
> Yes. The answer is rather simple: Change the checkout tenant mid-checkout. Since we have the same steps for every tenant and the only difference is the payment provider we haven't observed any loss of previously collected checkout data yet.
> 
> HTH, Marc

@wuw-marc sorry to reply on such an old thread but how did you change the tenant?
